ui/app/utils/properties/watch.js

```diff
@@ -24,6 +25,16 @@ export function watchRecord(modelName) {
           wait(throttle),
         ]);
       } catch (e) {
+        if (!ignore404) {
+          const statusIs404 = getWithDefault(e, 'errors', [])
+            .mapBy('status')
+            .includes('404');
+          const errorIsEmptyResponse = e.message.includes('response did not have any data');
+          if (statusIs404 || errorIsEmptyResponse) {
+            this.flashMessages.warning(`This ${modelName} no longer exists`, { sticky: true });
+          }
+        }
```

Comment:
I tried that out in [this branch](https://github.com/hashicorp/nomad/compare/f-ui/blocking-query-404s-flash...f-ui/blocking-query-404s-flash-evented), what do you think? There’s some awkwardness involving bundling the `modelName` along with the error because otherwise the error handler doesn’t know what to put in the flash message. Another approach would be to use the `args` on the task, as the model is passed in to `perform`, but given that the signature is `id` vs `model`, it seems questionable.

I also tried making it an [encapsulated task](http://ember-concurrency.com/docs/encapsulated-task) so it could contain `modelName` for future reference, but that seemed to require passing in the Ember Data store because `this` would therefore be the task instance and not the containing model 😞 

One significant problem is that the tests no longer pass:

![image](https://user-images.githubusercontent.com/43280/59933759-dcfc2880-93fe-11e9-98c4-c29c8e0a1151.png)

That could be partially addressed by ignoring the uncaught exception, hopefully in a more targeted way to avoid how that [burned me in the past](https://github.com/emberjs/ember.js/issues/12791#issuecomment-192318923).

BUT it’s not just that; the assertion that a flash message is displayed is also failing, despite me being able to see that it’s there 🤔 (that’s the inscrutable `failed` error 8 which I opened [this PR](https://github.com/hashicorp/nomad/pull/5869) to improve so we don’t need to write custom assertion failure messages)

I can hopefully address these problems but I wanted to check in on what you thought of the `modelName` weirdness and the general approach before sinking more time into it.

Reply:
this could perhaps be revived as it’s not a huge change and would still be nice to have, maybe the above weirdness is a non-problem now 🤞